cpp/tensorrt_llm/thop/fp8Quantize.cpp

```diff
@@ -65,6 +66,18 @@ std::tuple<at::Tensor, at::Tensor> fp8_quantize_1x128(at::Tensor const& self)
     mGemmRunner.fp8CS1x128(
         act_buffer, act_scale_buffer, reinterpret_cast<__nv_bfloat16 const*>(self.data_ptr()), n, m, stream);
 
+    // Post-process the scale tensor for sm100 gemm/moe kernel
+    if (tensorrt_llm::common::getSMVersion() == 100)
+    {
+        auto const num_n_blocks = (n + 127) / 128;
+        auto const act_scal_elesize = num_n_blocks * m_padded;
+        TORCH_CHECK(act_scal_elesize <= scaleFP8SF.numel(), "Scale tensor size mismatch. Expected at least ",
+            act_scal_elesize, " elements, got ", scaleFP8SF.numel());
+
+        // scaleFP8SF = scaleFP8SF[0:num_n_blocks, 0:m] // no 4-element alignment in blackwell
+        scaleFP8SF
+            = scaleFP8SF.slice(0, 0, act_scal_elesize).view({num_n_blocks, m_padded}).slice(1, 0, m).contiguous();
```

Comment:
`contiguous()` invokes a copy. Ideally, we should fix the kernel instead of using two ops on SM100. Can you add a TODO here?

Reply:
makes sense. Updated